core/src/main/java/org/elasticsearch/common/transport/NetworkExceptionHelper.java

```diff
@@ -35,7 +36,7 @@ public static boolean isConnectException(Throwable e) {
     }
 
     public static boolean isCloseConnectionException(Throwable e) {
-        if (e instanceof ClosedChannelException) {
+        if (e instanceof ClosedChannelException || e instanceof SocketException) {
```

Comment:
out of curiosity - what other SocketExceptions should be treated as the if the connection is closed (i.e., it should be closed if it isn't)?